Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -4846,6 +4849,131 @@ def panorama_override_vulnerability(threatid: str, vulnerability_profile: str, d
+
+    return http_request(
+        URL,
+        'GET',
+        params=params)
+
+
+def panorama_route_lookup(dest_ip: str, virtual_router=None):
+    """
+    Given the provided ip address, looks up the outgoing interface and zone on the firewall.
+    """
+    if not VSYS:
+        raise Exception("The 'panorama-route-lookup' command is only relevant for a Firewall instance.")
+
+    response = panorama_get_routes(virtual_router)
+    if 'entry' not in response['response']['result']:
+        raise Exception("No routes returned from the Firewall.")
+    else:
+        routes = response['response']['result']['entry']
+
+    ip_addr = ipaddress.ip_address(dest_ip)
+    current_match = None
+    matched_route = None
+    for route in routes:
+        subnet_raw = route['destination']
+
+        subnet = ipaddress.ip_network(subnet_raw)
+        # If the given IP address is in the subnet
+        if ip_addr in subnet:
+            # IF we haven't matched yet
+            if not current_match:
+                current_match = subnet
+                matched_route = route
+            # If this is a greater subnet
+            elif subnet.prefixlen > current_match.prefixlen:
+                current_match = subnet
+                matched_route = route
+
+    if matched_route:
+        return matched_route
+    else:
+        raise Exception("Route not found.")
+
+
+def panorama_route_lookup_command():
+    """
+    Gets the outgoing interface from the Palo Alto Firewall route table
+    Must be run against a firewall instance
+    """
+    dest_ip = demisto.args().get("dest_ip")
+    vr = demisto.args().get("virtual_router", None)
+    demisto.results(panorama_route_lookup(dest_ip, vr))
+
+
+def panorama_zone_lookup_command():
+    """
+    Gets the outgoing interface from the Palo Alto Firewall route table, and the list of interfaces
+    comparing the two
+    """
+    dest_ip = demisto.args().get("dest_ip")
+    vr = demisto.args().get("virtual_router", None)
+    route = panorama_route_lookup(dest_ip, vr)
+    if not route:
+        demisto.results(f"Could find a matching route to {dest_ip}.")
+        return
+
+    interface = route["interface"]
+    interfaces = panorama_get_interfaces()
+
+    r = {}
+    if "ifnet" in interfaces["response"]["result"]:
+        for entry in interfaces["response"]["result"]["ifnet"]["entry"]:
+            if entry["name"] == interface:
+                if "zone" in entry:
+                    r = {**entry, **route}
+
+    if r:
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': r,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': f'The IP {dest_ip} is in zone {r["zone"]}',
+            'EntryContext': {"Panorama.ZoneLookup(val.Name == obj.Name)": r}  # add key -> deleted: true
+        })
+        return r
+    else:
+        demisto.results(f"Could not map {dest_ip} to zone.")
```

Comment:
please use return_results method instead,
see this [doc](https://xsoar.pan.dev/docs/integrations/code-conventions#return_results)